ArchLinux: cmake "non-existent path" errors due to OpenCV_CONFIG_PATH

##### System information (version)

- OpenCV => 4.1.0
- Operating System / Platform => Arch Linux / amd64
- Compiler => gcc

##### Detailed description

While attempting to build [sdrangel](https://github.com/f4exb/sdrangel) (which has a dependency on OpenCV), I encounter a rather peculiar build error:

```
CMake Error in pluginssrv/channeltx/modatv/CMakeLists.txt:
  Imported target "opencv_calib3d" includes non-existent path

    "//include/opencv4"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

#### The problem seems to be [this line](https://github.com/opencv/opencv/blob/master/cmake/templates/OpenCVConfig.cmake.in#L79):
```
set(OpenCV_CONFIG_PATH "${CMAKE_CURRENT_LIST_DIR}")
```

On Arch this sets `OpenCV_CONFIG_PATH` to `/lib64/cmake/opencv4/`

Then the next line in OpenCVConfig.cmake is:
```
get_filename_component(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/../../../" REALPATH)
```
... which resolves to `/lib64/cmake/opencv4/../../../` which is simply `/`, which is clearly incorrect.

`/lib64` symlinks to `/usr/lib`, so a quick fix is to change:
```
set(OpenCV_CONFIG_PATH "${CMAKE_CURRENT_LIST_DIR}")
```
to
```
get_filename_component(OpenCV_CONFIG_PATH "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
```
which will resolve the `/lib64` symlink, correctly setting `OpenCV_CONFIG_PATH` to `/usr/lib/cmake/opencv4`

##### Steps to reproduce

On Arch Linux, attempt to build sdrangel-git from AUR.

--- a/cmake/templates/OpenCVConfig.cmake.in
+++ b/cmake/templates/OpenCVConfig.cmake.in
@@ -76,7 +76,7 @@ endif()
 
 # Extract the directory where *this* file has been installed (determined at cmake run-time)
 # Get the absolute path with no ../.. relative marks, to eliminate implicit linker warnings
-set(OpenCV_CONFIG_PATH "${CMAKE_CURRENT_LIST_DIR}")
+get_filename_component(OpenCV_CONFIG_PATH "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
 get_filename_component(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/@OpenCV_INSTALL_PATH_RELATIVE_CONFIGCMAKE@" REALPATH)
 
 # Search packages for host system instead of packages for target system.

